node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeAlerter.java

```diff
@@ -0,0 +1,447 @@
+package com.yahoo.vespa.hosted.provision.maintenance;
+
+import com.yahoo.config.provision.NodeResources;
+import com.yahoo.config.provision.NodeType;
+import com.yahoo.jdisc.Metric;
+import com.yahoo.log.LogLevel;
+import com.yahoo.vespa.hosted.provision.Node;
+import com.yahoo.vespa.hosted.provision.NodeRepository;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.yahoo.vespa.hosted.provision.node.Allocation;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * Performs analysis on the node repository to produce metrics that can be used for alerts if the repository is in an
+ * undesirable state.
+ * These metrics include:
+ * Spare host capacity, or how many hosts the repository can stand to lose without ending up in a situation where it's
+ * unable to find a new home for orphaned tenants.
+ * Overcommitted hosts, which tracks if there are any hosts whose capacity is less than the sum of its children's.
+ *
+ * @author mgimle
+ */
+public class NodeAlerter extends Maintainer {
```

Comment:
Change the class name to something different, as it doesn't actually alert anything. Perhaps something like  `CapacityReportMaintainer`